File: Printly-frontend/src/lib/api/tenant.ts
import apiClient from "./client";

export interface TenantProfileResponse {
	id: string;
	name: string;
	slug: string | null;
	address: string | null;
	phone: string | null;
	email: string | null;
	logo_url: string | null;
	is_active: boolean;
	created_at: string;
}

export interface TenantUpdateRequest {
	name?: string;
	phone?: string;
	address?: string;
	email?: string;
}

// TODO: Backend endpoint /tenants/me GET and PUT needed
export const tenantApi = {
	getProfile: async (): Promise<TenantProfileResponse> => {
		const response = await apiClient.get<TenantProfileResponse>("/tenants/me");
		return response.data;
	},
	updateProfile: async (
		data: TenantUpdateRequest,
	): Promise<TenantProfileResponse> => {
		const response = await apiClient.put<TenantProfileResponse>(
			"/tenants/me",
			data,
		);
		return response.data;
	},
};

File: Printly-backend/app/schemas/tenant.py
from pydantic import BaseModel, Field, ConfigDict
from uuid import UUID
from datetime import datetime


class TenantProfileResponse(BaseModel):
    id: UUID
    name: str
    slug: str | None
    address: str | None
    phone: str | None
    email: str | None
    logo_url: str | None
    is_active: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TenantUpdateRequest(BaseModel):
    name: str | None = Field(None, max_length=200)
    address: str | None = None
    phone: str | None = Field(None, max_length=20)
    email: str | None = Field(None, max_length=255)
    logo_url: str | None = Field(None, max_length=500)